### ISSUE TITLE
typescript rollup config with node 22 fails with "Unexpected identifier 'assert'"

### ISSUE THREAD

**Issue by michaelglass** [rollup/plugins#1762]
- Rollup Plugin Name: typescript
- Rollup Plugin Version: 11.1.6
- Rollup Version: 4.14.0
- Operating System (or Browser): Mac OS 14.6.1
- Node Version: 22.4.1
- Link to reproduction _(⚠️ read below)_: https://github.com/michaelglass/rollup-typescript-repro

### Expected Behavior
on Node V22, the keyword `assert` has been replaced with `with`, 
when my `rollup.config.ts` includes an import using `with`

```
import pkg from "./package.json" with { type: "json" };
```

`npx rollup -c --pluginConfig typescript`
will run rollup

### Actual Behavior
I get the error, 

```sh
[!] SyntaxError: Unexpected identifier 'assert'
    at compileSourceTextModule (node:internal/modules/esm/utils:337:16)
    at ModuleLoader.moduleStrategy (node:internal/modules/esm/translators:164:18)
    at callTranslator (node:internal/modules/esm/loader:439:14)
    at ModuleLoader.moduleProvider (node:internal/modules/esm/loader:445:30)
    at async ModuleJob._link (node:internal/modules/esm/module_job:106:19)
 ```

I get the same error when passing the tsconfig manually to rollup, e.g.
`npx rollup -c --configPlugin typescript={target:\"esnext\"\,module:\"Preserve\"}`

### Additional Information

clone the repo above to see a small reproduction

```sh
git clone git@github.com:michaelglass/rollup-typescript-repro.git
cd rollup-typescript-repro
direnv allow # if you're using nix / direnv, or nix develop
npm install
npm run rollup-ts
```

**Comment by shellscape** [comment]
This is a TypeScript + Node semantic language issue. I don't see how Rollup has an impact.

**Comment by michaelglass** [comment]
`npx tsc` will create a node-22 compatible rollup config, and like I mentioned in the issue, passing in the compiler config as command line arguments results in the same failure.

Perhaps something in rollup or the typescript plugin that is ignoring the tsconfig file, or loading it late? (Or perhaps I'm doing something wrong!)

**Comment by shellscape** [comment]
I'm really not sure, but rollup doesn't have any bearing on tsc running. that's going to be an issue with tsc. you've probably got a typescript config issue somewhere, bc import assertions are only supported in extremely specific config combos. 

but that's not under the scope of rollup. I'd recommend heading to StackOverflow or running the code and ts config through Claude.

**Comment by michaelglass** [comment]
yeah you could be right, but it seems like my tsconfig is fine, considering that using `tsc` directly produces a node-22 compatible rollup config but running `npx rollup -c --pluginConfig typescript` seems to be producing / using a node-22 incompatible rollup config.
